src/main/java/net/szum123321/elytra_swap/mixin/PlayerFallCapture.java

```diff
@@ -37,44 +37,42 @@ protected PlayerFallCapture(EntityType<? extends LivingEntity> type, World world
         fallingHandler(heightDifference, onGround, landedState, landedPosition);
         super.fall(heightDifference, onGround, landedState, landedPosition);
     }
 
-    private void fallingHandler(double heightDifference, boolean onGround, BlockState landedState, BlockPos landedPosition){
-        if((Object)this instanceof ServerPlayerEntity) {
+    private void fallingHandler(double heightDifference, boolean onGround, BlockState landedState, BlockPos landedPosition) {
+        if ((Object) this instanceof ServerPlayerEntity) {
             PlayerEntity player = (PlayerEntity) (Object) this;
 
-            if(ElytraSwap.inventoryController.doesPlayerHaveElytra(player)){
-                if(!PlayerSwapDataHandler.get(player))
+            if (ElytraSwap.inventoryController.doesPlayerHaveElytra(player)) {
+                if (!PlayerSwapDataHandler.get(player))
                     return;
 
-                if(!onGround && !player.isClimbing() && !player.isSwimming()){
+                if (!onGround && !player.isClimbing() && !player.isSwimming()) {
                     if (PlayerSwapDataHandler.get(player) && heightDifference < 0 && getFallHeight(landedPosition) > 5 && (ServerSidePacketRegistry.INSTANCE.canPlayerReceive(player, ElytraSwap.DUMMY_PACKAGE) || ElytraSwap.config.noModPlayersHandlingMethod > 0)) {
                         ElytraSwap.inventoryController.replaceChestPlateWithElytra(player);
                         setFlag(7, true);    // thanks to this line you do not have to press space in order to start gliding
                     }
-                }else if(PlayerSwapDataHandler.get(player) && (ServerSidePacketRegistry.INSTANCE.canPlayerReceive(player, ElytraSwap.DUMMY_PACKAGE) || ElytraSwap.config.noModPlayersHandlingMethod > 0)){
+                } else if (PlayerSwapDataHandler.get(player) && (ServerSidePacketRegistry.INSTANCE.canPlayerReceive(player, ElytraSwap.DUMMY_PACKAGE) || ElytraSwap.config.noModPlayersHandlingMethod > 0)) {
                     ElytraSwap.inventoryController.replaceElytraWithChestPlate(player);
                     setFlag(7, false);
                 }
             }
         }
     }
 
-    private int getFallHeight(BlockPos currentPosition){
-        int height = currentPosition.getY();
-
-        while(!world.getBlockState(new BlockPos(currentPosition.getX(), height, currentPosition.getZ())).getMaterial().isSolid() && height > -1){
-            height--;
-        }
-
-        if(height <= -1){
-            ElytraSwap.LOGGER.info("WTF! Why are you trying to glide below bedrock?");
-            return -Math.abs(currentPosition.getY()) * 2; //even if you would fly below bedrock it should work :)
+    private float getFallHeight(BlockPos currentPosition) {
+        try (BlockPos.PooledMutable temp = BlockPos.PooledMutable.get(currentPosition.getX(), Math.max(currentPosition.getY(), world.getHeight()), currentPosition.getZ())) {
```

Comment:
You are getting max of world height and player's position and than going down form that but this creates one big problem. What if there are some blocks above player? Than this function returns negative number which is obviously smaller than 5. Please correct that and I'll merge and publish new version.